Assets/Scripts/Battlefield/StateBehaviors/PlayerTurnStateBehavior.cs

```diff
@@ -3,9 +3,12 @@
 using UnityEngine;
 using SwordAndBored.Battlefield;
 using UnityEngine.EventSystems;
+using SwordAndBored.Battlefield.AStar;
 
 public class PlayerTurnStateBehavior : StateMachineBehaviour
 {
+    AStar star = new AStar();
+    DisplayPath show = new DisplayPath();
```

Comment:
This and pretty much all the code in OnStateEnter should be replace with a Movement System abstraction such as

```
MovementSystem aStarIsBorn = new AStarMovementSystem(grid);
```
and in OnStateUpdate()
```
aStarIsBorn.MoveAgent(agent, startPoint, endPoint);
```
or something similar. That is easier to read and debug and change if need be.